pkg/apis/duck/v1alpha1/credentials.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"regexp"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
+)
+
+const (
+	DefaultSecretName = "google-cloud-key"
+	defaultSecretKey  = "key.json"
+)
+
+var (
+	validation_regexp = regexp.MustCompile(`^[A-Za-z0-9-]+@[A-Za-z0-9-]+\.iam.gserviceaccount.com$`)
```

Comment:
got this error when creating one invalid one from gcloud:

```ERROR: (gcloud.iam.service-accounts.create) argument NAME: Bad value [%alpha-svc-acct%]: Service account name must be between 6 and 30 characters (inclusive), must begin with a lowercase letter, and consist of lowercase alphanumeric characters that can be separated by hyphens.```

we might need to enforce size here, and starting with lower case...

Reply:
won't block this PR on it... just add it to the backlog, maybe create a new issue?

Reply:
I was trying to create a service account for testing some other things. 

```
export SVC_ACCT=%alpha-svc-acct%
gcloud iam service-accounts create ${SVC_ACCT}
```

if you play around with it, you can get the limits right...

Reply:
Thank you, Nacho! I created issue #669